src/utils/Client.util.ts

```diff
@@ -117,3 +120,60 @@ export function getContentType(data: object | string): string {
     return "application/json";
   }
 }
+
+/**
+ * Configure the entries for bulk publish request.
+ * If entryIDs are missing, generate UUIDs for them.
+ * If contentTypes are missing, infer them based on the data using {@link getContentType}.
+ *
+ * @param entries entries to configure
+ * @returns configured entries
+ */
+export function configureBulkPublishEntries(entries: PubSubBulkPublishEntry[]): PubSubBulkPublishEntry[] {
+  for (let i = 0; i < entries.length; i++) {
+    const entry = entries[i];
+    // If entryID is missing, generate a UUID for it.
+    if (!entry.entryID) {
+      entry.entryID = randomUUID();
+    }
+    // If contentType is missing, infer it from the data.
+    if (entry.data && !entry.contentType) {
+      entry.contentType = getContentType(entry.data);
+    }
+  }
+  return entries;
+}
+
+export type BulkPublishApiResponse = {
+  statuses: {
```

Comment:
Same as earlier, let's use a sub type that we can use as array here `BulkPublishApiResponseStatus`